demo/app/lib/views/credential_preview.dart

```diff
@@ -35,142 +35,81 @@ class CredentialPreviewState extends State<CredentialPreview> {
-      var credentialClaimsData = data['credential_displays'][0]['claims'] as List;
-      list = credentialClaimsData.map<CredentialPreviewData>((json) => CredentialPreviewData.fromJson(json)).toList();
-    return list;
-  }
-  Widget listViewWidget(List<CredentialPreviewData> credPrev) {
-    return ListView.builder(
-        itemCount: credPrev.length,
-        scrollDirection: Axis.vertical,
-        shrinkWrap: true,
-        itemBuilder: (context, position) {
-          //TODO Ignoring the photo value for now due to extremely long text need to render in a separate issue-881
-          return (credPrev[position].label != "Photo") ? Card(
-            child: ListTile(
-              title: Text(
-                credPrev[position].label,
-                style: const TextStyle(fontSize: 14, fontFamily: 'SF Pro', fontWeight: FontWeight.w400, color: Color(0xff6C6D7C))
-              ),
-              subtitle: Text(
-                credPrev[position].value,
-                style: const TextStyle(
-                    fontSize: 16,
-                    color: Color(0xff190C21),
-                    fontFamily: 'SF Pro',
-                    fontWeight: FontWeight.normal),
-              ),
-            ),
-          ):
-          Card(
-            child: Row(
-              crossAxisAlignment: CrossAxisAlignment.center,
-              children: [
-                Expanded(
-                  child: Padding(
-                    padding: const EdgeInsets.all(24.0),
-                    child: Column(
-                      mainAxisAlignment: MainAxisAlignment.spaceAround,
-                      crossAxisAlignment: CrossAxisAlignment.start,
-                      mainAxisSize: MainAxisSize.min,
-                      children: [
-                        Text(credPrev[position].label, style: const TextStyle(fontSize: 14, fontFamily: 'SF Pro', fontWeight: FontWeight.w400, color: Color(0xff6C6D7C)),),
-                      ],
-                    ),
-                  ),
-                ),
-                Flexible(
-                  fit: FlexFit.tight,
-                  child: Padding(
-                    padding: const EdgeInsets.all(24.0),
-                    child: Column(
-                      mainAxisAlignment: MainAxisAlignment.spaceAround,
-                      crossAxisAlignment: CrossAxisAlignment.end,
-                      mainAxisSize: MainAxisSize.min,
-                      children: <Widget>[
-                    Image.memory(const Base64Decoder().convert(credPrev[position].value.split(',').last), width: 80, height: 80,),
-                     ],
-                    ),
-                  ),
-                ),
-              ],
-            ),
-          );
-        });
-  }
   @override
   Widget build(BuildContext context) {
-    Map<String, dynamic> issuer = jsonDecode(widget.credentialDisplay);
+    Map<String, dynamic> issuer = jsonDecode(widget.credentialData.credentialDisplayData);
```

Comment:
It better not have heavy methods like jsonDecode inside the build method. It is called often when some data changes. 
you can override  void initState() method, and call jsonDecode there

Reply:
Moved into initistate function as suggested